pkg/server/backuprestoreserver.go

```diff
@@ -197,25 +192,26 @@ func (b *BackupRestoreServer) runServer(ctx context.Context, restoreOpts *brtype
 		}
 	}
 
-	var memberPeerURL string
-
 	m := member.NewMemberControl(b.config.EtcdConnectionConfig)
-	err := retry.OnError(retry.DefaultBackoff, errors.AnyError, func() error {
+	if err := retry.OnError(retry.DefaultBackoff, errors.AnyError, func() error {
 		cli, err := etcdutil.NewFactory(*b.config.EtcdConnectionConfig).NewCluster()
 		if err != nil {
 			return err
 		}
-		memberPeerURL, err = m.UpdateMemberPeerURL(ctx, cli)
-		if err != nil {
+		defer cli.Close()
+
+		if err := m.UpdateMemberPeerURL(ctx, cli); err != nil {
 			return err
 		}
 		return nil
-	})
-	if err != nil {
+	}); err != nil {
 		b.logger.Errorf("failed to update member peer url: %v", err)
 	}
 
-	peerURLTLSEnabled := b.isPeerURLTLSEnabled(memberPeerURL)
+	peerURLTLSEnabled, err := miscellaneous.IsPeerURLTLSEnabled()
```

Comment:
Can you consider the error thrown by `m.UpdateMemberPeerURL()` before this function is called?
If `m.UpdateMemberPeerURL()` returns an error then I don't think we should call `miscellaneous.IsPeerURLTLSEnabled()`

Reply:
`IsPeerURLTLSEnabled` is now independent of `UpdateMemberPeerURL`.
>Can you consider the error thrown by m.UpdateMemberPeerURL() before this function is called?
If m.UpdateMemberPeerURL() returns an error then I don't think we should call miscellaneous.IsPeerURLTLSEnabled()

that's the whole purpose of this PR, if restoration is currently going on then definitely `UpdateMemberPeerURL` will fail but that shouldn't affect our decision of checking peerURLTLSEnabled or not.
And TBH I want to remove this whole `UpdateMemberPeerURL` logic from here as I think we don't need a backward compatibility now. right ?
Anyway `UpdateMemberPeerURL`is called after restoration.

Reply:
ok, I got your concern. Let me explain this:
First we are not doing anything with this PeerURL which I read from config file, I'm just interested in checking whether it is `http` or `https` , which I get from the config file the same way [UpdateMemberPeerURL](https://github.com/gardener/etcd-backup-restore/blob/f69cd4deac013ed0cd5a0949a488be376fe9e87c/pkg/member/member_control.go#L188-L193) get PeerURL to update.
Now take each scenarios:
**case 1:** now when restoration fails and [UpdateMemberPeerURL](https://github.com/gardener/etcd-backup-restore/blob/f69cd4deac013ed0cd5a0949a488be376fe9e87c/pkg/snapshot/restorer/restorer.go#L124-L131) didn't get execute.
then there will be no annotation added to member lease as member lease updation will get stop by backup-restore as etcd is not running.

**case 2:** replica=1, when restoration succeeds but [UpdateMemberPeerURL](https://github.com/gardener/etcd-backup-restore/blob/f69cd4deac013ed0cd5a0949a488be376fe9e87c/pkg/snapshot/restorer/restorer.go#L124-L131) fails
then it won't matter as it is the case of single node, no peer communication is required.

**case 3:** scale-up=0->1->3, when restoration succeeds in `0->1` but [UpdateMemberPeerURL](https://github.com/gardener/etcd-backup-restore/blob/f69cd4deac013ed0cd5a0949a488be376fe9e87c/pkg/snapshot/restorer/restorer.go#L124-L131) fails
then as I mentioned earlier that I'm not doing anything with this PeerURL which I get from the config file the same way [UpdateMemberPeerURL](https://github.com/gardener/etcd-backup-restore/blob/f69cd4deac013ed0cd5a0949a488be376fe9e87c/pkg/member/member_control.go#L188-L193) get PeerURL to update. Suppose `UpdateMemberPeerURL` succeeds then it would be the same peerURL on that basis I have calculated whether the TLS enable or not.
But `UpdateMemberPeerURL` fails, so  moving from `replica 1->replicas 3` won't be possible as this member will not be able to join the cluster, as it requires to have a peer communication. then I guess we have bigger problem which is not limited to annotation. This Problem is out-of-scope of this PR.

Reply:
Can you give me the edge case which is handled by the current code but not handled by this PR and where this can cause issue as may be I don't have much of a insight on this whole TLS not enabled -> TLS enabled scenario ?
If you have new edge case which is currently not handled by current code as well as by this PR then I would suggest open a new issue describing the edge case. Hope that's fine.